### PR TITLE
Adjusting the readme to be for v2, rename ToEventMessage to ToMessage

### PR DESCRIPTION
**Pull request by n3wscott**
- adjusting the readme to be for v2
- rename ToEventMessage to ToMessage

Signed-off-by: Scott Nichols <snichols@vmware.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,126 +10,121 @@
...
-under heavy development and will not be following the same semver rules as
-`pkg/cloudevents`. If a release is required to ship changes to `pkg/bindings`, a
-bug fix release will be issued (x.y.z+1).
+For protocols that do not support responses from send (Requester interface),
+`Client.Request` will throw an error.
 
-We will target ~2 months of development to release v2 of this SDK with an end
-date of March 27, 2020. You can read more about the plan for SDK v2 in the
-[SDK v2 planning doc](./docs/SDK_v2.md).
+**v2.0.0-preview1:**
```
Comment (slinkydeveloper) [inline README.md]: Don't forget to update it to preview2 :wink:

File: README.md [hunk 2]
```diff
@@ -10,126 +10,121 @@
...
-	if err != nil {
-		log.Fatalf("failed to create client, %v", err)
-	}
-	log.Fatal(c.StartReceiver(context.Background(), Receive));
-}
+bytes, err := json.Marshal(event)
 ```
 
-Creating a minimal CloudEvent in version 1.0:
+The SDK can help bind this to a specific protocol, in this case HTTP Request:
```
Comment (slinkydeveloper) [inline README.md]: Before this sentence, I would add just a quick explanation of what is `binding.Message`, something like:

> binding.Message is an abstraction to read a protocol specific message as a CloudEvent, without fully parsing it to `cloudevents.Event`, in order to allow optimized message passing between different protocols.

File: README.md [hunk 3]
```diff
@@ -10,126 +10,121 @@
...
+req, _ = nethttp.NewRequest("POST", "http://localhost", nil)
+err = http.WriteRequest(context.TODO(), msg, req, nil)
+// ...check error.
+
+// Then use req:
+resp, err := http.DefaultClient.Do(req)
 ```
 
-Sending a cloudevents.Event via the HTTP Transport with Binary v1.0 encoding:
+The SDK can let you stay at the `Event` level if you want to use the `Client`.
```
Comment (slinkydeveloper) [inline README.md]: This one should live before the binding.Message example

File: README.md (hunk 1 above)
Comment (n3wscott) [reply README.md]: this was for preview1 that we cut last week. preview2 is for this week.

File: README.md (hunk 3 above)
Comment (n3wscott) [reply README.md]: I reworded

**Pull request merged**